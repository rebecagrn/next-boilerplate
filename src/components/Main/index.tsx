const Main = () => (
  <main>
    <h1>Nextjs Boilerplate</h1>
  </main>
)

export default Main
