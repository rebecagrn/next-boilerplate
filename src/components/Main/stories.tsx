import Main from '.'
import { Story, Meta } from '@storybook/react'
import { ComponentProps } from 'styled-components/node_modules/@types/react'

export default {
  title: 'Main',
  component: Main
} as Meta

const Template: Story<ComponentProps<typeof Main>> = (args) => (
  <Main {...args} />
)

export const Basic = Template.bind({})
Basic.args = {
  title: 'Nextjs Boilerplate',
  description: 'TypeScript, ReactJS, NextJS and Styled Components'
}
