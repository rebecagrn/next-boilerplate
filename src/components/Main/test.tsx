import { screen, render } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('it should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /nextjs boilerplate/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
