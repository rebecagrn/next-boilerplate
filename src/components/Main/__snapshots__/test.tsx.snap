// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> it should render the heading 1`] = `
<main>
  <h1>
    Nextjs Boilerplate
  </h1>
</main>
`
